### DISCUSSION TITLE
Ouster SDK Lifecycle Policies

### DISCUSSION THREAD

**Discussion by kairenw** [ouster-lidar/ouster_example#532] [Announcements]
On all platforms and languages, we align Ouster SDK end-of-life support with the publicly communicated standard support for that platform. We will periodically update as new platforms and languages are added to Ouster SDK support.

For sensor firmwares, we also wish to give advance notice of dropped support for new versions of the SDK.

We wish to stress that for all users, should you be unable to upgrade your platform, Python, or C++ version for any reason, you can always use an older release by checking out the appropriate commit/tag on Github or installing an older Python package. If you are an Ouster customer who anticipates buying new sensors on new firmware but have platform constraints that conflict with the above, please contact your FAE or Sales representative.

We will update this with new platforms, Pythons, FWs etc as they are added.

### Ubuntu
[https://wiki.ubuntu.com/Releases](https://wiki.ubuntu.com/Releases
)
| Ubuntu LTS version | End of support |
| -----------------------------| ---------------------- |
| Ubuntu 18.04 | October 2023 |
| Ubuntu 20.04 | April 2025 |
| Ubuntu 22.04 | April 2027 |

### Debian
[https://wiki.debian.org/LTS](https://wiki.debian.org/LTS)
| Debian version | End of support |
| -------- | ------- |
| Debian 10 | roughly  June 2024 |
| Debian 11 | roughtly June 2026 |

### MacOS
Major versions of MacOS are generally supported for 3 years after release. As such:
| MacOS version | End of support |
| --- | --- |
| macOS 11 (Big Sur) | likely October 2023 |
| macOS 12 (Montery) | likely October 2024 |
| macOS 13 (Ventura) | likely October 2025 |

### Windows
[https://learn.microsoft.com/en-us/lifecycle/faq/windows#windows-11](https://learn.microsoft.com/en-us/lifecycle/faq/windows#windows-11)
| Windows version | End of support |
| --- | --- |
| Windows 10 | October 2025 |
| Windows 11 | currently unknown |

### ROS
ROS 1 will have limited support towards the end of 2023. 
ROS 2 foxy will receive limited support towards the end of 2023 since the distribution is past its end of life date and we are providing support as a courtesy to suers.

In general, ROS communications are made through the [ouster-ros](https://github.com/ouster-lidar/ouster-ros) repo, so please follow there for announcements.

### Python
https://devguide.python.org/versions/
| Python version | End of ouster-sdk support |
| ---- |---- |
| Python 3.7 | October 2023 |
| Python 3.8 | October 2024 |
| Python 3.9 | October 2025 |
| Python 3.10 | October 2026 |
| Python 3.11 | October 2027 |

### C++
C++ does not go out of support. However, we will likely begin to use C++17 features within the next 6-12 months, so newer versions of the SDK will not be compatible with projects which need to use C++14. There are currently no plans to move to C++20 within the next 2 years. This means that Windows users building the C++ SDK must use VS Studio 17 or later.

## Sensor Firmwares

We are dropping support for FW 2.0 in November 2023, as previously announced. There is no firm support policy for now, but it is likely we will drop FW 2.1 sometime next year (2024). 

## User courtesy per previous announcements

As per [our previous announcement](https://github.com/ouster-lidar/ouster_example/discussions/497), as a courtesy for users we are maintaining Ubuntu 18.04, Python 3.7, and macOS 11 support past their respective end-of-life/end-of-supports. This is an exception to the rule going forward.

**Comment by kinotus** [comment]
Hi, any update on the "Sensor firmwares"-section regarding 2024 and 2025? Also interested if Ubuntu 24.04 LTS will be officially supported.

**Reply by twslankard** [reply]
@kinotus 

We're announcing end of support for FW 2.2 and 2.3 by end of June 2025. There may be other versions deprecated in 2025, so stay tuned. Please let me know if any questions or concerns arise.

Official announcement is here: https://github.com/ouster-lidar/ouster-sdk/issues/638

Best Regards,

Tom Slankard
Staff Engineer
Ouster Inc.
